### ISSUE TITLE
Curious problem with wildcard match

### ISSUE THREAD

**Issue by yoshi314** [matteocorti/check_ssl_cert#31]
I am seeing an error like this : 

`SSL_​CERT CRITICAL *.domain.​com: invalid CN ('*.domain.​com' does not match '*.domain.​com')`

I am not sure how to tackle this issue.

**Comment by matteocorti** [comment]
How did you call (options) the plugin?

**Comment by yoshi314** [comment]
/usr/local/lib/nagios/plugins/check_ssl_cert -H 193.138.208.171 -w 10 -c 5 -r /tmp -n *.connector-b2b.com --ignore-ocsp --ignore-sig-alg

**Comment by matteocorti** [comment]
One of the problem is that the certificate specifies http://ilo-ca.abs.assecobs.pl/cpd-lublin.crt as ISSUER_URI. The link is invalid

```
wget http://ilo-ca.abs.assecobs.pl/cpd-lublin.crt
--2016-12-06 09:12:19--  http://ilo-ca.abs.assecobs.pl/cpd-lublin.crt
Resolving ilo-ca.abs.assecobs.pl (ilo-ca.abs.assecobs.pl)... failed: nodename nor servname provided, or not known.
wget: unable to resolve host address ‘ilo-ca.abs.assecobs.pl’
```

**Comment by matteocorti** [comment]
The certificate is identified as self signed

```
SSL_CERT CRITICAL *.abstore.biz: Cannot verify certificate, self signed certificate|days=421;;;;
```

Since you have a self-signed certificate in the chain

```
verify depth is 6
depth=2 C = PL, ST = lubelskie, L = Lublin, O = Asseco Busines Solutions S.A., OU = CPD Lublin, CN = assecobs.pl, emailAddress = datacenter.lub@assecobs.pl
verify error:num=19:self signed certificate in certificate chain
verify return:1
depth=2 C = PL, ST = lubelskie, L = Lublin, O = Asseco Busines Solutions S.A., OU = CPD Lublin, CN = assecobs.pl, emailAddress = datacenter.lub@assecobs.pl
verify return:1
depth=1 O = AssecoBS CPD CA, CN = Certyfikat Instancji CA AssecoBS CPD CA
verify return:1
depth=0 C = PL, ST = LUBLIN, L = LUBLIN, O = Asseco Business Solutions S.A., CN = *.abstore.biz, emailAddress = datacenter.lub@assecobs.pl
verify return:1
DONE
```

If I add `--selfsigned` I get the expected result

```
./check_ssl_cert -H 193.138.208.171 --ignore-ocsp --ignore-sig-alg --selfsigned 
SSL_CERT OK - X.509 self signed certificate for '*.abstore.biz' from 'Certyfikat Instancji CA AssecoBS CPD CA' valid until Jan 31 11:52:27 2018 GMT (expires in 421 days)|days=421;;;; 
```

And with the CN (which does not match)

```
./check_ssl_cert -H 193.138.208.171 --ignore-ocsp --ignore-sig-alg --selfsigned -n '*.connector-b2b.com'
SSL_CERT CRITICAL *.connector-b2b.com: invalid CN ('*.connector-b2b.com' does not match '*.connector-b2b.com')|days=281;;;;
```

**Comment by yoshi314** [comment]
The CA system is internal, so that might explain the selfsigned chain issue on your end. This is a non-issue here.

I still do not understand the CN mismatch, because i am not seeing any ( '*.connector-b2b.com' does not match '*.connector-b2b.com' ) . Or does the existence of self-signed cert trigger the error about CN mismatch?

**Comment by matteocorti** [comment]
Something is strange.

In the certificate I see

```
        Subject: OU=Domain Control Validated, CN=*.connector-b2b.com
```

but when connecting with a browser I get a certificate for `*.abstore.biz`

**Comment by matteocorti** [comment]
Actually it seems a bug.

**Comment by matteocorti** [comment]
Fixed with 1.36.1
